packages/workers-shared/asset-worker/src/handler.ts

```diff
@@ -203,7 +203,9 @@ export const canFetch = async (
 			flagIsEnabled(
 				configuration,
 				SEC_FETCH_MODE_NAVIGATE_HEADER_PREFERS_ASSET_SERVING
-			) && request.headers.get("Sec-Fetch-Mode") === "navigate"
+			) &&
+			request.headers.get("Sec-Fetch-Mode") === "navigate" &&
+			!configuration.has_static_routing
```

Comment:
Can we also test this? It would be great to see this as:

1. a test making a `Sec-Fetch-Mode: navigate` request who's outcome changes based off of the presence of unmatched static routing, and
2. a test which uses the `env.ASSETS.unstable_canFetch(request)` API, again showing different results based off of the presence/absence of static routing.

Reply:
there are tests for 1. currently, but located in https://github.com/cloudflare/workers-sdk/pull/9304 (they're fixtures, so they depended on the wrangler implementation). i can add a unit test here

for 2., that seems like the same test as 1. - there's no additional behavior we'd be testing. wouldn't complain if you're just looking to get a more unit-style test here as compared to 1.